File: PalindromeNumber/palindromeNumber.cpp
/*
 * Determine whether an integer is a palindrome.
 * Do this without extra space.
 *
 */

#include <iostream>
#include <cstdlib>

class Solution {
  public:
    bool isPalindrome(int x) {
      int reversedNum = 0;
      int y = abs(x);
      while(y > 0) {
        reversedNum = (reversedNum*10) + (y%10);
        y /= 10;
      }
      if( abs(x) == reversedNum ) {
        return true;
      } else {
        return false;
      }
    }
};

int main(int argc, char* argv[]) {
  Solution sol;
  int number = atoi(argv[1]);
  bool result;
  const char *resultString;
  result = sol.isPalindrome(number);
  if(result) {
    resultString = "True";
  } else {
    resultString = "False";
  }
  
  std::cout << "Solution: " << resultString << std::endl;

  return 0;
}

File: PalindromeNumber/palindromeNumber.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

"""
Determine whether an integer is a palindrome.
Do this without extra space.
"""

from sys import argv

class Solution(object):
  def isPalindrome(self, x):
    """
    :type x: int
    :rtype: bool
    """
    if str(abs(x))[::-1] == str(x):
      return True
    else:
      return False

sol = Solution()

if len(argv) == 2:
  print "Solution: " + str(sol.isPalindrome(int(argv[1])))
else:
  print "Wrong number of arguments\n"
